src/lua/TPTScriptInterface.cpp

```diff
@@ -119,23 +119,23 @@ ValueType TPTScriptInterface::testType(String word)
 
 int TPTScriptInterface::parseNumber(String str)
 {
-	String::value_type const *stringData = str.c_str();
```

Comment:
The `c_str` is actually better because you can index one element past the end of string and that would return `0` which wouldn't compare equal to any of the chars below. `stringIter[0]` on an empty string however is undefined behavior.